packages/graphql/src/schema-model/attribute/Field.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Neo4jGraphQLSchemaValidationError } from "../../classes/Error";
+import type { Annotation, Annotations } from "../annotation/Annotation";
+import { annotationToKey } from "../annotation/Annotation";
+
+export class Field {
```

Comment:
Fields represent GraphQL fields in the root Operation types; Query, Mutation and Subscription. They are different to Attributes, because an attribute in our schema model implies that it is backed by a database node property. Field is a lot more lightweight, just needed for basic authentication for custom Cypher queries.

Reply:
I guess the way we were looking at it:

- `Operation`/`Field` === "GraphQL only" things
- `Entity`/`Attribute` === mapped to some Neo4j entity

As such, `Operation`/`Field` have a much more limited feature set, with hardly any directives being usable with them, for example.